### PR TITLE
WIP Set Resource Based Naming in dual-stack Subnets

### PR DESCRIPTION
**Pull request by johngmyers**

### CODE REVIEW

File: upup/pkg/fi/cloudup/awstasks/launchtemplate_target_api.go [hunk 1]
```diff
@@ -55,6 +55,10 @@ func (t *LaunchTemplate) RenderAWS(c *awsup.AWSAPITarget, a, e, changes *LaunchT
 				Ipv6AddressCount:         t.IPv6AddressCount,
 			},
 		},
+		PrivateDnsNameOptions: &ec2.LaunchTemplatePrivateDnsNameOptionsRequest{
+			EnableResourceNameDnsAAAARecord: fi.Bool(*t.IPv6AddressCount > 0),
```
Comment (olemarkus) [inline upup/pkg/fi/cloudup/awstasks/launchtemplate_target_api.go:59]: Should we not enable A records too?

**Review by olemarkus — COMMENTED** [review]
There is also an attribute on the subnet to enable RBN. Not sure we need to set that one as well.

File: upup/pkg/fi/cloudup/awstasks/launchtemplate_target_api.go (hunk 1 above)
Comment (johngmyers) [reply upup/pkg/fi/cloudup/awstasks/launchtemplate_target_api.go:59]: They aren't default?

Comment (olemarkus) [reply upup/pkg/fi/cloudup/awstasks/launchtemplate_target_api.go:59]: They are for ipv6 native, but not sure about dual stack

**Comment by johngmyers** [comment]
Perhaps it can be set either on subnet or launch template? I suspect we'd be better off setting it on subnet, so users could customize how they like on shared subnets.

**Comment by johngmyers** [comment]
There's an attribute on the `Subnet` but I don't see any API for setting it.

**Comment by hakman** [comment]
> Perhaps it can be set either on subnet or launch template? I suspect we'd be better off setting it on subnet, so users could customize how they like on shared subnets.

LaunchTemplate is easier exactly because of shared subnets. We can add whatever settings we need and keep them consistent, without modifying or failing if shared subnets are misconfigured.

**Comment by johngmyers** [comment]
I wouldn't say we "need" to set the `HostnameType`, it's just that RBN is a better default. It would be better for us to say "If you want IPBN, create your own shared subnets" than either add an option or prohibit them.

**Comment by hakman** [comment]
Creating new subnets might be inconvenient. I agree that IPBN is a better default in general.

**Comment by johngmyers** [comment]
Looks like CCM can't deal with RBN.

**Comment by olemarkus** [comment]
yeah. I expected that.
Probably an easy fix. Maybe this time just use hostname as node name so that we don't have to pass in hostname override ...

**Comment by johngmyers** [comment]
Here's the culprit:
https://github.com/kubernetes/cloud-provider-aws/blob/a1590733fac851b3a27d351c9c80e9b2bf8d6f7e/pkg/providers/v1/aws.go#L5174-L5182

**Comment by johngmyers** [comment]
Filed kubernetes/cloud-provider-aws#285

**Pull request closed without merging**